### ISSUE TITLE
Environment plugin does not work if it is a scoped package

### ISSUE THREAD

**Issue by hbarcelos** [eslint/eslint#12190]
**Tell us about your environment**

* **ESLint Version:** 6.2.2
* **Node Version:** 12.8.0
* **npm Version:** 6.10.2

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

<details>
<summary>Attempt 1</summary>

```json
{
  "env": {
    "truffle/truffle": true
  },
  "plugins": [
    "@4cadia/eslint-plugin-truffle"
  ]
}
```
</details>

<details>
<summary>Attempt 2</summary>

```json
{
  "env": {
    "@4cadia/eslint-plugin-truffle/truffle": true
  },
  "plugins": [
    "@4cadia/eslint-plugin-truffle"
  ]
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I have created a little plugin to define `truffle` global variables:

```js
module.exports = {
  environments: {
    truffle: {
      globals: {
        artifacts: 'readonly',
        contract: 'readonly',
        assert: 'readonly',
        web3: true,
      },
    },
  },
};
```
Currently, I am using `npm link` to use it as a local package. The gotcha is that it was supposed to be a scoped package: `@4cadia/eslint-plugin-truffle`.

Then in the actual repo which depends upon the plugin, I run it as:
```bash
eslint test/Indexer.test.js
```

**What did you expect to happen?**
ESLint should use the plugin and not report any errors in the file as described in the [docs](https://eslint.org/docs/developer-guide/working-with-plugins#environments-in-plugins).

**What actually happened? Please include the actual, raw output from ESLint.**
I get the following error:

```
> Error: .eslintrc:
>         Environment key "truffle/truffle" is unknown
> 
```
The same happens if I use the `"@4cadia/eslint-plugin-truffle/truffle"` env in the configuration file instead.

**IMPORTANT:** However, if I drop the `@4cadia` scope in the package name and publish/link it as `eslint-plugin-truffle`, the code from Attempt 1 works as expected.

**Are you willing to submit a pull request to fix this bug?**

I am not familiar with the ESLint code base. If someone could please point me to a place to start, I would be happy to contribute.

**Comment by platinumazure** [comment]
Hi @hbarcelos, thanks for the issue.

I think we should definitely support scoped packages here. We'll mark this as accepted if we can recreate the issue.

**Comment by mysticatea** [comment]
Actually, this is the correct behavior.

To specify a plugin env, you have to use the naming convention as same as plugin rules'.

```jsonc
{
  "env": {
    "@4cadia/truffle/truffle": true // not "@4cadia/eslint-plugin-truffle/truffle"
  },
  "rules": {
    "@4cadia/truffle/a-rule": "error" // not "@4cadia/eslint-plugin-truffle/a-rule"
  },
  "plugins": [
    "@4cadia/truffle"
  ]
}
```

**Comment by mysticatea** [comment]
The naming convention:

- `eslint-plugin-foo` → `foo/a-rule`
- `@foo/eslint-plugin` → `@foo/a-rule`
- `@foo/eslint-plugin-bar` → `@foo/bar/a-rule`

**Comment by hbarcelos** [comment]
Thank's a lot for the clarification @mysticatea. It works exactly as you described.

I believe we should probably make this clear in the [docs](https://eslint.org/docs/developer-guide/working-with-plugins), because there is not much information there about working with plugins from scoped package names.

**Comment by kaicataldo** [comment]
Closing this issue as the question has been answered.
